File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionFindTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionFindTest extends BaseTest {
    @Test
    void findTheTruth() {
        assertTransformation(fromJson("""
[0, [], "a"]"""), fromJson("""
{
  "$$find": "$", "by": "##current"
}
"""), "a");

        // explicit boolean (non js style)
        assertTransformation(fromJson("""
["a", "1", "true"]"""), fromJson("""
{
  "$$find": "$", "by": "$$boolean:##current"
}
"""), "true");
    }
    @Test
    void nameThatStartsWithB() {
        assertTransformation(fromJson("""
[{"name":"alice"}, {"name":"ann"}, {"name":"Bob"}]"""), fromJson("""
{
  "$$find": "$", "by": "$$test('(?i)^b'):##current.name"
}
"""), fromJson("""
{"name":"Bob"}"""));
    }

    @Test
    void inline() {
        assertTransformation(fromJson("[0, [], \"a\"]"), "$$find(##current):$", "a");
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionFind.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionFind", () => {
  test("findTheTruth", async () => {
    await assertTransformation(
      [0, [], "a"],
      {
        $$find: "$",
        by: "##current",
      },
      "a",
    );

    // explicit boolean (non js style)
    await assertTransformation(
      ["a", "1", "true"],
      {
        $$find: "$",
        by: "$$boolean:##current",
      },
      "true",
    );
  });

  test("nameThatStartsWithB", async () => {
    await assertTransformation(
      [{ name: "alice" }, { name: "ann" }, { name: "Bob" }],
      {
        $$find: "$",
        by: "$$test('(?i)^b'):##current.name",
      },
      { name: "Bob" },
    );
  });

  test("inline", async () => {
    await assertTransformation([0, [], "a"], "$$find(##current):$", "a");
  });
});
